orc8r/cloud/go/services/service_registry/servicers/kubernetes_servicer.go

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright 2020 The Magma Authors.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package servicers
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"magma/orc8r/lib/go/protos"
+
+	"golang.org/x/net/context"
+	corev1types "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+const (
+	partOfLabel                    = "app.kubernetes.io/part-of"
+	serviceRegistryNamespaceEnvVar = "SERVICE_REGISTRY_NAMESPACE"
+	grpcPortName                   = "grpc"
+	httpPortName                   = "http"
+	serviceNameDelimiter           = "-"
+)
+
+type KubernetesServiceRegistryServicer struct {
+	client    corev1.CoreV1Interface
+	namespace string
+}
+
+// NewKubernetesServiceRegistryServicer creates a new service registry servicer
+// that is backed by Kubernetes.
+func NewKubernetesServiceRegistryServicer(k8sClient corev1.CoreV1Interface) (*KubernetesServiceRegistryServicer, error) {
+	namespaceEnvValue := os.Getenv(serviceRegistryNamespaceEnvVar)
+	if len(namespaceEnvValue) == 0 {
+		return nil, fmt.Errorf("%s was not provided as an environment variable", serviceRegistryNamespaceEnvVar)
+	}
+	return &KubernetesServiceRegistryServicer{
+		client:    k8sClient,
+		namespace: namespaceEnvValue,
+	}, nil
+}
+
+// ListAllServices returns the service name of all services in the registry.
+func (s *KubernetesServiceRegistryServicer) ListAllServices(ctx context.Context, req *protos.Void) (*protos.ListAllServicesResponse, error) {
+	ret := &protos.ListAllServicesResponse{}
+	orc8rListOption := metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("%s=%s", partOfLabel, "orc8r"),
+	}
+	svcList, err := s.client.Services(s.namespace).List(orc8rListOption)
+	if err != nil {
+		return ret, err
+	}
+	for _, svc := range svcList.Items {
+		splitSvcName := strings.SplitAfterN(svc.Name, serviceNameDelimiter, 2)
+		svcNameIndex := len(splitSvcName) - 1
+		ret.Services = append(ret.Services, splitSvcName[svcNameIndex])
```

Comment:
same as for the other servicer, factor this out and add some context as to what you're doing here